### ISSUE TITLE
Jottacloud not working for me

### ISSUE THREAD

**Issue by agent0404** [ncw/rclone#3320]
#### What is the problem you are having with rclone?
I have just signed up for Jottacloud for the sole purpose of using it with rclone only. If I can't get this to work I wont be using Jottacloud and will probably just delete my account.

I cannot get `rclone config` to set up my Jottacloud remote. At first I thought it was because I had 2FA.. so I disabled it.. but it still doesn't work.

I tried many different combinations when it asks for my username. It's expecting the email I used to sign up for Jottacloud with.. right?

#### What is your rclone version (output from `rclone version`)
```
rclone v1.48.0
- os/arch: linux/amd64
- go version: go1.12.5
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Arch

####  Which cloud storage system are you using? (eg Google Drive)
Jottacloud

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
```
rclone config
```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
```
Your Jottacloud password is only required during setup and will not be stored.
password:
2019/07/01 Failed to get resource token: HTTP error 400 (400 Bad Request) returned body: "{\"error\":\"invalid_grant\",\"error_description\":\"The provided authorization grant (e.g., authorization code, resource owner credentials) or refresh token is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client.\"}"
```

> The provided authorization grant (e.g., authorization code, resource owner credentials) or refresh token is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client.

**Comment by ncw** [comment]
> I tried many different combinations when it asks for my username. It's expecting the email I used to sign up for Jottacloud with.. right?

correct

I just tried making a new remote with `rclone config` and it worked fine

```
$ rclone-v1.48 --config /tmp/j.conf config
2019/07/02 09:59:26 NOTICE: Config file "/tmp/j.conf" not found - using defaults
No remotes found - make a new one
n) New remote
s) Set configuration password
q) Quit config
n/s/q> n
name> jottacloud
Type of storage to configure.
Enter a string value. Press Enter for the default ("").
Choose a number from below, or type in your own value
[snip]
14 / JottaCloud
   \ "jottacloud"
[snip]
Storage> jottacloud
** See help for jottacloud backend at: https://rclone.org/jottacloud/ **

User Name:
Enter a string value. Press Enter for the default ("").
user> me@example.com
Edit advanced config? (y/n)
y) Yes
n) No
y/n> n
Remote config

Do you want to create a machine specific API key?

Rclone has it's own Jottacloud API KEY which works fine as long as one only uses rclone on a single machine. When you want to use rclone with this account on more than one machine it's recommended to create a machine specific API key. These keys can NOT be shared between machines.

y) Yes
n) No
y/n> y
Your Jottacloud password is only required during setup and will not be stored.
password:

Do you want to use a non standard device/mountpoint e.g. for accessing files uploaded using the official Jottacloud client?

y) Yes
n) No
y/n> y
Please select the device to use. Normally this will be Jotta
Choose a number from below, or type in an existing value
 1 > Jotta
Devices> 1
Please select the mountpoint to user. Normally this will be Archive
Choose a number from below, or type in an existing value
 1 > Archive
 2 > Shared
 3 > Sync
Mountpoints> 3
--------------------
[jottacloud]
type = jottacloud
user = me@example.com
client_id = XXX
client_secret = XXX
token = {"access_token":"XXX","token_type":"bearer","refresh_token":"XXX","expiry":"2019-07-02T11:00:00.781183416+01:00"}
device = Jotta
mountpoint = Sync
--------------------
y) Yes this is OK
e) Edit this remote
d) Delete this remote
y/e/d> y
Current remotes:

Name                 Type
====                 ====
jottacloud           jottacloud

e) Edit existing remote
n) New remote
d) Delete remote
r) Rename remote
c) Copy remote
s) Set configuration password
q) Quit config
e/n/d/r/c/s/q> q

$ rclone-v1.48 --config /tmp/j.conf lsd jottacloud:
          -1 2019-07-02 10:00:28        -1 dir
          -1 2019-07-02 10:00:28        -1 path
          -1 2019-07-02 10:00:28        -1 secret
          -1 2019-07-02 10:00:28        -1 testchars
```

**Comment by agent0404** [comment]
Well I've even tried it on multiple different machines. Using the install script on a VPS to get the latest version

I even changed the password to make it simpler (using only letters and numbers because before it was a super long randomly generated blob with symbols)

I'm not typing my password wrong during the rclone config because I've tried copying and pasting and using autotype feature of my password manager directly into the terminal

Oh well it's not too important I just figured I would try out Jottacloud's free 5GB plan. Thanks anyways.. I use rclone with many other cloud services and it works great!

**Comment by buengese** [comment]
@agent0404 Are you still interested in helping me debug this? I'm interested in your specific setup because you are 3 user reporting difficulties logging into Jottacloud and so far I've not been able to reproduce this in any of those cases. If you are could you run the setup with ``rclone config -v -v --dump bodies`` and provide me with the full output. Of course make sure to remove any passwords / tokens appearing in the output or use a throwaway account.

**Comment by agent0404** [comment]
I'm more than happy to help any way I can

```
2019/07/04 00:21:31 DEBUG : Trying to register device 'removed'
2019/07/04 00:21:31 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2019/07/04 00:21:31 DEBUG : HTTP REQUEST (req 0xc00049e000)
2019/07/04 00:21:31 DEBUG : POST /auth/v1/register?device_id=removed HTTP/1.1
Host: api.jottacloud.com
User-Agent: rclone/v1.48.0
Content-Length: 0
Authorization: XXXX
Content-Type: application/x-www-form-urlencoded
Accept-Encoding: gzip

2019/07/04 00:21:31 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2019/07/04 00:21:32 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2019/07/04 00:21:32 DEBUG : HTTP RESPONSE (req 0xc00049e000)
2019/07/04 00:21:32 DEBUG : HTTP/1.1 200 OK
Content-Length: 87
Content-Type: application/json
Date: Thu, 04 Jul 2019 00:21:32 GMT
Strict-Transport-Security: max-age= 31536000
X-Id: removed
X-Resp: 11

{"client_id":"id","client_secret":"id"}
2019/07/04 00:21:32 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2019/07/04 00:21:32 DEBUG : Saving config "client_id" = "id" in section "jotta" of the config file
2019/07/04 00:21:32 DEBUG : Saving config "client_secret" = "id" in section "jotta" of the config file
2019/07/04 00:21:32 DEBUG : Got clientID 'id' and clientSecret 'id'
Your Jottacloud password is only required during setup and will not be stored.
password:
2019/07/04 00:21:42 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2019/07/04 00:21:42 DEBUG : HTTP REQUEST (req 0xc000032400)
2019/07/04 00:21:42 DEBUG : POST /auth/v1/token?client_id=removed&client_secret=removed&grant_type=PASSWORD&password=email%40domain.com%09removed&username=email%40domain.com HTTP/1.1
Host: api.jottacloud.com
User-Agent: rclone/v1.48.0
Content-Length: 0
Content-Type: application/x-www-form-urlencoded
Accept-Encoding: gzip

2019/07/04 00:21:42 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2019/07/04 00:21:42 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2019/07/04 00:21:42 DEBUG : HTTP RESPONSE (req 0xc000032400)
2019/07/04 00:21:42 DEBUG : HTTP/1.1 400 Bad Request
Content-Length: 285
Content-Type: application/json
Date: Thu, 04 Jul 2019 00:21:42 GMT
Strict-Transport-Security: max-age= 31536000
X-Id: removed
X-Resp: 123

{"error":"invalid_grant","error_description":"The provided authorization grant (e.g., authorization code, resource owner credentials) or refresh token is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client."}
2019/07/04 00:21:42 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2019/07/04 00:21:42 Failed to get resource token: HTTP error 400 (400 Bad Request) returned body: "{\"error\":\"invalid_grant\",\"error_description\":\"The provided authorization grant (e.g., authorization code, resource owner credentials) or refresh token is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client.\"}"
```

Is the `password` field in the request supposed to have the email in it as well? Even though there is a `username` field following it

```
&password=email%40domain.com%09removed&username=email%40domain.com
```

**Comment by buengese** [comment]
no that's definitely not supposed to happen and it would explain your problem. I'll look into what might cause this.

**Comment by buengese** [comment]
I really don't see anything in the code that should be able to cause this. So my current theory that rclone is somehow getting confused when reading console input and thus reads your username twice. We can verify this using this rclone binary [Link](https://drive.google.com/file/d/1wPutatieyYKRllMl4jbOSD5U9slkgFWn/view?usp=sharing). It should mirror your password back to you. If the output contains your email I'd like to your shell and terminal and also if your pasting any of your inputs.

**Comment by agent0404** [comment]
Well this is embarrassing.. it totally slipped my mind that when I use the autotype feature of my password manager that it is typing out the email-TAB-password

But that doesn't explain why the first two times I tried setting it up I was getting the same error even though I had copied and pasted my password manually. Maybe the first time was because of 2FA. then I disabled it. And the second time (that I copied and pasted the password instead of using autotype) was because of my password being super long and with weird symbols

Anyways, it's working now. I'm sure that definitely made you scratch your head while looking at the code trying to figure that out. Whoops haha
